### PR TITLE
chore: fix `Service` doc links in tower-layer

### PR DESCRIPTION
**Pull request by Folyd**
My bad. The [`Service`]: https://docs.rs/tower/trait.Service.html is a broken link. It should be https://docs.rs/tower/*/tower/trait.Service.html.

### CODE REVIEW

File: tower-layer/src/lib.rs [hunk 1]
```diff
@@ -90,7 +90,7 @@ pub use self::{
 /// is also decoupled from client or server concerns. In other words, the same
 /// log middleware could be used in either a client or a server.
 ///
-/// [`Service`]: https://docs.rs/tower/trait.Service.html
+/// [`Service`]: https://docs.rs/tower/*/tower/trait.Service.html
```
Comment (davidbarsky) [inline tower-layer/src/lib.rs:93]: Same:

```suggestion
/// [`Service`]: tower_service::Service
```

File: tower-layer/src/lib.rs [hunk 2]
```diff
@@ -14,7 +14,7 @@
 //!
 //! A middleware implements the [`Layer`] and [`Service`] trait.
 //!
-//! [`Service`]: https://docs.rs/tower/trait.Service.html
+//! [`Service`]: https://docs.rs/tower/*/tower/trait.Service.html
```
Comment (davidbarsky) [inline tower-layer/src/lib.rs:17]: Thanks for fixing this! Can these also be switched to intradoc links? I think the following should work:

```
//! [`Service`]: tower_service::Service
```

**Review by davidbarsky — CHANGES_REQUESTED** [review]
Thanks

File: tower-layer/src/lib.rs (hunk 1 above)
Comment (Folyd) [reply tower-layer/src/lib.rs:93]: Thanks for reviewing. I tried but failed. Any suggestions?

```log
$ cargo doc --no-deps --all-features 
error: unresolved link to `tower_service::Service`
  --> tower-layer/src/lib.rs:17:18
   |
17 | //! [`Service`]: tower_service::Service
   |                  ^^^^^^^^^^^^^^^^^^^^^^ no item named `tower_service` in scope
   |
note: the lint level is defined here
  --> tower-layer/src/lib.rs:8:9
   |
8  | #![deny(broken_intra_doc_links)]
   |         ^^^^^^^^^^^^^^^^^^^^^^
   = note: `#[deny(rustdoc::broken_intra_doc_links)]` implied by `#[deny(broken_intra_doc_links)]`

error: aborting due to previous error

error: could not document `tower-layer`

warning: build failed, waiting for other jobs to finish...
error: build failed
```

Same error with `cargo doc --all-features`.

**Comment by davidpdrsn** [comment]
@Folyd do you wanna drive this home?

**Comment by Folyd** [comment]
> @Folyd do you wanna drive this home?

Hi @davidpdrsn. I mistakenly modified the link of `[Service]` in my previous PR(#578). This PR simply rectifies it.

However, the `[Service]: tower_service::Service` isn't working, so I change it to `[Service]: https://docs.rs/tower/*/tower/trait.Service.html`.

**Review by LucioFranco — APPROVED** [review]

**Pull request merged**